pipeline/train/train.py

```diff
@@ -0,0 +1,413 @@
+                    "valid-reset-stalled": None,
+                    "sync-sgd": None,
+                    "quiet-translation": None,
+                    "overwrite": None,
+                    "keep-best": None,
+                    "tsv": None,
+                }
+            ),
+            *extra_args,
+        ]
+
+    def run_training(self):
+        """
+        OpusTrainer pipes augmented data into Marian. Marian handles the training and
+        outputs the progress as in its log. The final part of the pipeline is the log
+        parser which parses the streamed logs and reports the results to W&B.
+        """
+        run_pipeline(
+            [
+                [
+                    # OpusTrainer controls the marian commands.
+                    *self.get_opustrainer_cmd(),
+                    *self.get_marian_cmd(),
+                ],
+                get_log_parser_command(),
+            ],
+            pipe_stderr=True,
+            logger=logger,
+        )
+
+        shutil.copy(
+            self.artifacts / f"model.npz.best-{self.best_model_metric}.npz",
+            self.artifacts / f"final.model.npz.best-{self.best_model_metric}.npz",
+        )
+        shutil.copy(
+            self.artifacts / f"model.npz.best-{self.best_model_metric}.npz.decoder.yml",
+            self.artifacts / f"final.model.npz.best-{self.best_model_metric}.npz.decoder.yml",
+        )
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        # Preserves whitespace in the help text.
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+
+    parser.add_argument(
+        "--model_type",
+        type=ModelType,
+        choices=ModelType,
+        required=True,
+        help="The type of model to train",
+    )
+    parser.add_argument(
+        "--training_type",
+        type=TrainingType,
+        choices=TrainingType,
+        help="Type of teacher training",
+    )
+    parser.add_argument("--gpus", type=str, required=True, help="GPUs to use")
+    parser.add_argument(
+        "--marian_dir",
+        type=Path,
+        required=True,
+        help="Path to Marian binary directory. This allows for overriding to use the browser-mt fork.",
+    )
+    parser.add_argument("--workspace", type=str, required=True, help="Workspace directory")
+    parser.add_argument("--src", type=str, help="Source language")
+    parser.add_argument("--trg", type=str, help="Target language")
+    parser.add_argument(
+        "--train_set_prefixes",
+        type=str,
+        help="Comma separated prefixes to datasets for curriculum learning",
+    )
+    parser.add_argument("--validation_set_prefix", type=str, help="Prefix to validation dataset")
+    parser.add_argument("--artifacts", type=Path, help="Where to save the model artifacts")
+    parser.add_argument("--vocab", type=Path, help="Path to vocab file")
+    parser.add_argument("--best_model_metric", type=str, help="Best model metric")
```

Comment:
it's in fact one of expected values, so it should be an enum of supported metrics